### ISSUE TITLE
Angular 11: @ngtools/webpack : forkTypeChecker intermittently behaves as though directTemplateLoading is true

### ISSUE THREAD

**Issue by Rush** [angular/angular-cli#19407]
# 🐞 Bug report

### Is this a regression?
It became worse in Angular 11 / @ngtools/webpack 11.

### Description

I use pug for my angular templates:
```ts
@Component({
  selector: 'app-page',
  templateUrl: './app-page.component.pug',
  styleUrls: ['./app-page.component.scss'],
  animations: [
```
As I understand `@ngtools/webpack` rewrites it to `templateUrl: require('./app-page.component.pug')` and it uses my configured loaders.

Anyway, the problem is that when I build my codebase with `webpack-dev-server` it intermittently behaves as though it doesn't recognize my loaders when validating types. It only affects type checking as the code works perfectly. This is the output:

```
ℹ ｢wds｣: Project is running at http://0.0.0.0:8080/
ℹ ｢wds｣: webpack output is served from http://localhost:8080/
ℹ ｢wds｣: Content not from webpack is served from ./public-built/
(node:14486) DeprecationWarning: loaderUtils.parseQuery() received a non-string value which can be problematic, see https://github.com/webpack/loader-utils/issues/56
parseQuery() will be replaced with getOptions() in the next major version of loader-utils.
(Use `node --trace-deprecation ...` to show where the warning was created)
ℹ ｢wdm｣: Compiled with warnings.
ℹ ｢wdm｣: Compiling...

Error: modules/flowcharts/app/components/code-viewer-split.component.pug:108:1 - error NG5002: Unexpected character "EOF" (Do you have an unescaped "{" in your template? Use "{{ '{' }}") to escape it.)

108 
    

  modules/flowcharts/app/components/code-viewer-split.component.ts:17:16
    17   templateUrl: './code-viewer-split.component.pug',
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component CodeViewerSplitComponent.
modules/flowcharts/app/components/code-viewer-split.component.pug:35:19 - error NG5002: Invalid ICU message. Missing '}'.

35                 | {{ realTimeCollaborator.userInfo.displayName }}

..... <MORE_MESSAGES_HERE>

  modules/flowcharts/app/components/diagram-viewer.component.ts:91:16
    91   templateUrl: './diagram-viewer.component.pug',
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component DiagramViewerComponent.

ℹ ｢wdm｣: Compiled with warnings.
```
Then I go my dev page everything is perfect.

What is weird is that restarting webpack a few times usually gets rid of the error. The intermittent part is what puzzles me. To make the issue occur 100% of the times I can simply do:
```ts
new AngularCompilerPlugin({
    tsConfigPath: IS_PRODUCTION ? 'tsconfig.prod.json' : 'tsconfig.dev.json',
    entryModule: path.join(__dirname, 'src/app/app.module#AppModule'),
    forkTypeChecker: false,
    directTemplateLoading: true, # the errors above happen due to this!
  })
```

The workaround is to set `forkTypeChecker: false` and keep `directTemplateLoading: false`.

Therefore my conclusion is that something makes the forked type checker load templates directly. Any ideas how to debug it?

## 🔬 Minimal Reproduction
I haven't yet found standalone reproduction steps but I welcome any tips on how to debug it ...

## 🔥 Exception or Error
I pasted above. Looking for tips on how to capture the error in the code. Where to put console logs?

## 🌍 Your Environment
Not running CLI - just `@ngtools/webpack`
```
» cat yarn.lock |grep ngtools/webpack
    "@ngtools/webpack" "11.0.1"
"@ngtools/webpack@11.0.1", "@ngtools/webpack@^11.0.1":
  resolved "https://registry.yarnpkg.com/@ngtools/webpack/-/webpack-11.0.1.tgz#029aaff16d76fbfdf06422dde4c83f3b79c34215"
```

**Anything else relevant?**
It's occurring at least both on Linux and Mac

**Comment by alan-agius4** [comment]
Duplicate https://github.com/angular/angular-cli/issues/18290

**Comment by Rush** [comment]
@alan-agius4 got it. Any idea why this is intermittent then?

**Comment by alan-agius4** [comment]
Likely it’s because the forked type checker is  only used after the first successful run.

NB: the new Ivy only plugin which is still experimental and will be available 11.1 will not use fork type checking.
